internal/security/vaultclient/makereadcloser.go

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package vaultclient
+
+import (
+	"io"
+	"io/ioutil"
+)
+
+// makeReadCloser will turn an an io.Reader into an io.ReadCloser
+// if the underlying object does not already support io.ReadCloser
+func makeReadCloser(reader io.Reader) io.ReadCloser {
```

Comment:
this seems to be duplicated from the other PR?

Reply:
Yes, but it breaks the dependency and it is only a 3 line function..

Reply:
Re-introduced dependency as https://github.com/edgexfoundry/edgex-go/pull/1788 is merged now.